src/YamlWriter/CommandHelpYamlWriter.cs

```diff
@@ -0,0 +1,270 @@
+using Microsoft.PowerShell.PlatyPS.Model;
+
+namespace Microsoft.PowerShell.PlatyPS.YamlWriter
+{
+    /// <summary>
+    /// Write the CommandHelp object to a file in yaml format.
+    /// </summary>
+    internal sealed class CommandHelpYamlWriter : CommandHelpWriterBase
+    {
+        /// <summary>
+        /// Initialize the writer with settings.
+        /// </summary>
+        /// <param name="settings">Settings needs for the yaml writer.</param>
+        internal CommandHelpYamlWriter(CommandHelpWriterSettings settings)
+        {
+            string path = settings.DestinationPath;
+
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+            else
+            {
+                _filePath = path;
+                _encoding = settings.Encoding;
+                sb = Constants.StringBuilderPool.Get();
+            }
+        }
+
+        internal override void WriteMetadataHeader(CommandHelp help, Hashtable? metadata = null)
+        {
+            sb.AppendLine("metadata:"); // Constants.YamlHeader);
+            sb.AppendLine($"  external help file: {help.ModuleName}-help.xml");
+            sb.AppendLine($"  Module Name: {help.ModuleName}");
+            sb.AppendLine($"  online version: {help.OnlineVersionUrl}");
+            sb.AppendLine($"  title: {help.Title}");
+            sb.AppendLine($"  {Constants.SchemaVersionYaml}");
+
+            if (metadata is not null)
+            {
+                foreach (DictionaryEntry item in metadata)
+                {
+                    sb.AppendFormat("  {0}: {1}", item.Key, item.Value);
+                }
+            }
+        }
+
+        internal override void WriteTitle(CommandHelp help)
+        {
+            sb.AppendLine($"title: {help.Title}");
+        }
+
+        internal override void WriteSynopsis(CommandHelp help)
+        {
+            sb.AppendFormat("{0} '{1}'\n", Constants.SynopsisYamlHeader, help.Synopsis);
+        }
+
+        internal override void WriteSyntax(CommandHelp help)
+        {
+            sb.AppendLine(Constants.SyntaxYamlHeader);
+
+            if (help.Syntax?.Count > 0)
+            {
+                foreach (SyntaxItem item in help.Syntax)
+                {
+                    if (item.IsDefaultParameterSet)
+                    {
+                        var syntaxString = item.ToSyntaxString(Constants.DefaultSyntaxYamlTemplate);
+                        if (syntaxString is not null)
+                        {
+                            foreach(var line in syntaxString.Split(Constants.LineSplitter, StringSplitOptions.RemoveEmptyEntries))
+                            {
+                                if (line != "```")
+                                {
+                                    if (line.StartsWith("-"))
+                                    {
+                                        sb.AppendFormat("{0}\n", line);
```

Comment:
Would a `sb.AppendLine()` after the `sb.AppendFormat()` be better than and `\n` to avoid `\r\n` v/s `\n` issues.